dask/dataframe/optimize.py

```diff
@@ -32,3 +36,57 @@ def optimize(dsk, keys, **kwargs):
     )
     dsk, _ = cull(dsk, keys)
     return dsk
+
+
+def optimize_read_parquet_getitem(dsk):
+    # find the keys to optimze
+    from .io.parquet.core import ParquetSubgraph
+
+    read_parquets = [k for k, v in dsk.layers.items() if isinstance(v, ParquetSubgraph)]
+
+    layers_dict = dsk.layers.copy()
+
+    for k in read_parquets:
+        columns = set()
+
+        for dep in dsk.dependents[k]:
+            block = dsk.layers[dep]
+            if not isinstance(block, Blockwise):
+                return dsk
+
+            if len(block.dsk) != 1:
+                # TODO
+                return dsk
+
+            if list(block.dsk.values())[0][0] != operator.getitem:
+                return dsk
+
+            block_columns = block.indices[1][0]  # like ('B', None)
+            if isinstance(block_columns, str):
+                block_columns = [block_columns]
+
+            block_columns = set(block_columns)
+            columns |= block_columns
```

Comment:
Maybe combine these into one line?

```suggestion
            columns |= set(block_columns)
```